forge-core/src/main/java/forge/card/CardType.java

```diff
@@ -788,6 +802,20 @@ public boolean sharesCardTypeWith(final CardTypeView ctOther) {
         return false;
     }
 
+    public boolean sharesAllCardTypesWith(final CardTypeView ctOther) {
+        if (ctOther == null) {
+            return false;
+        }
+        List<String> ctOtherTypes = new ArrayList<>();
+        ctOther.getAllTypes().forEach(ctOtherTypes::add);
+        for (final String type : getAllTypes()) {
```

Comment:
The naming might have confused you but if you check above you'll see this only refers to the coreTypes
so you should just compare both cards amounts for that (so avoid extra list)

Reply:
I just looked at the rules and you are correct. I updated everything and removed what wasn't needed.

[Changes](https://github.com/Card-Forge/forge/pull/6257/commits/1f7a9197a1b09468ccfd32faf8352372d2757394)

(Sorry about the accidental force push!)